flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/api/TableUtils.java

```diff
@@ -100,7 +101,7 @@
 	 * for external output.
 	 */
 	private static TableSchema buildNewTableSchema(Table table) {
-		TableSchema oldSchema = table.getSchema();
+		TableSchema oldSchema = TableSchemaUtils.getPhysicalTableSchema(table.getSchema());
```

Comment:
I think we shouldn't change this line. The schema of `Table` may contains generated columns, but the generated column should also be outputed.